src/updater.ts

```diff
@@ -26,15 +26,45 @@ function installUpdate(): void {
     autoUpdater.quitAndInstall();
 }
 
-function pollForUpdates(): void {
+// Workaround for Squirrel.Mac wedging auto-restart if latest check for update failed
+// From https://github.com/vector-im/element-web/issues/12433#issuecomment-1508995119
+async function safeCheckForUpdate(): Promise<void> {
+    if (process.platform === "darwin") {
+        const feedUrl = autoUpdater.getFeedURL();
+        // On Mac if the user has already downloaded an update but not installed it and
+        // we check again and no additional new update is available the app ends up in a
+        // bad state and doesn't restart after installing any updates that are downloaded.
+        // To avoid this we check manually whether an update is available and call the
+        // autoUpdater.checkForUpdates() when something new is there.
```

Comment:
```suggestion
        // To avoid this we check manually whether an update is available and only call
        // autoUpdater.checkForUpdates() when something new is there.
```